Explore consensus cell types in brain and CNS tumors

allyhawkins opened the pull request:
### Purpose/implementation Section
#### Please link to the GitHub issue that this pull request addresses.

Closes #1005 

#### What is the goal of this pull request?

Here I'm summarizing the findings for consensus cell types in all samples that have a brain/cns diagnosis. This includes samples from `SCPCP000001`, `SCPCP000002`, `SCPCP000009`, `SCPCP000010`, and `SCPCP000021`. I did not include the samples that belong to multiplexed libraries. 

#### Briefly describe the general approach you took to achieve this goal.

- The overall flow of the notebook is similar to the osteo notebook, which means a lot of the setup code is copied from that notebook. First we look at the top cell types across all samples and then facet that by project id and subdiagnosis. I also look at the total immune cell classification and then specifically T cell populations vs. macrophages/ monocyte lineage. 
- As with osteo, there's variation across all the samples in terms of how many cells are classified and generally this is not project specific. I think the most interesting thing is looking at the diagnosis groups, since there appears to be a difference in T cell infiltrate and macrophage infiltrate in LGG vs. HGG. Obviously this is all qualitative, but I do think this could be something to pursue for showing in the paper. 
- For looking at sub diagnosis, I looked at the most represented diagnoses first and then in the last plot I grouped everything so it's either LGG or HGG.

Note that in working on this I moved the functions I used for plotting in the osteo notebook to their own script to source in. 

#### If known, do you anticipate filing additional pull requests to complete this analysis module?

Yes. 

### Results

#### What types of results does your code produce (e.g., table, figure)?

Here's a copy of the rendered report: 
[04-brain-and-cns-consensus-celltypes.nb.html.zip](https://github.com/user-attachments/files/18759443/04-brain-and-cns-consensus-celltypes.nb.html.zip)


#### What is your summary of the results?

As mentioned above I think the most interesting thing is the qualitative difference between T cell and macrophage infiltrate. It looks like Glioblastoma have a higher percentage of T cells and much less macrophages while the other types tend to mostly have a higher percentage of macrophages. This trend appears to be more generally divide HGG and LGG too, but again this is qualitative. 


### Author checklists

_Check all those that apply._
_Note that you may find it easier to check off these items after the pull request is actually filed._


#### Analysis module and review

- [ ] This analysis module [uses the analysis template and has the expected directory structure](https://openscpca.readthedocs.io/en/latest/contributing-to-analyses/analysis-modules/).
- [x] The analysis module `README.md` has been updated to reflect code changes in this pull request.
- [x] The analytical code is documented and contains comments.
- [ ] Any results and/or plots this code produces have been added to your S3 bucket for review.

#### Reproducibility checklist

- [ ] Code in this pull request has been added to the GitHub Action workflow that runs this module.
- [ ] The dependencies required to run the code in this pull request have been added to the analysis module `Dockerfile`.
- [ ] If applicable, the dependencies required to run the code in this pull request have been added to the analysis module conda `environment.yml` file.
- [ ] If applicable, R package dependencies required to run the code in this pull request have been added to the analysis module `renv.lock` file.

jaclyn-taroni commented on analyses/cell-type-consensus/exploratory-notebooks/04-brain-and-cns-consensus-celltypes.Rmd:
```diff
@@ -0,0 +1,403 @@
+Based on this let's make the following groups: 
+
+- Low-grade glioma: Any samples labeled with "Low-grade glioma" or are part of `SCPCP000002` which are categorized as LGG tumors. 
+- High-grade glioma: Any samples labeled with "High-grade glioma" or are part of `SCPCP000001` which are categorized as HGG tumors. 
```
This should [include Diffuse midline glioma](https://www.cancer.gov/rare-brain-spine-tumor/tumors/diffuse-midline-gliomas) as well ([another source](https://radiopaedia.org/articles/diffuse-midline-glioma-h3-k27-altered?lang=us)).

jaclyn-taroni commented on analyses/cell-type-consensus/exploratory-notebooks/04-brain-and-cns-consensus-celltypes.Rmd:
```diff
@@ -0,0 +1,403 @@
+  "macrophage", 
+  "monocyte",
+  "mononuclear phagocyte", 
+  "dendritic cell"
```
Could you only get a myeloid dendritic cell out of the consensus labels? <https://pmc.ncbi.nlm.nih.gov/articles/PMC5032838/>

jaclyn-taroni commented on analyses/cell-type-consensus/exploratory-notebooks/04-brain-and-cns-consensus-celltypes.Rmd:
```diff
@@ -0,0 +1,403 @@
+  stringr::str_subset("T")
+
+# macro/mono lineage 
+macro_mono_cells <- c(
```
You could probably call this myeloid and include granulocytes with little to no impact on the results?

jaclyn-taroni commented on analyses/cell-type-consensus/exploratory-notebooks/utils/plotting-functions.R:
```diff
@@ -0,0 +1,47 @@
+      title = plot_column
+    )
+  
+}
```
```suggestion
}

```

jaclyn-taroni reviewed (commented):
The overall plotting strategy looks good, but I have some comments/questions about various categorizations.

jashapiro reviewed (approved):
This looks good to me, with @jaclyn-taroni's comments addressed. One little additional comment.

allyhawkins commented:
Thanks @jaclyn-taroni and @jashapiro. Based on your comments, I added "Diffuse midline glioma" to the HGG group and then I adjusted the macro/mono group, which I'm now labeling as myeloid. In doing that I ended up removing "dendritic cell", since one of the possible dendritic cells is of lymphoid lineage (see [`plasmacytoid dendritic cell` or pDCs](https://www.ebi.ac.uk/ols4/ontologies/cl/classes/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FCL_0000784?lang=en)). Note that I checked and only one of the cells across all the samples used here was classified as pDC instead of DC, but I still kept it off. I did add "neutrophils" and "myeloid leukocyte" to the myeloid group, but I didn't add granulocytes (https://github.com/AlexsLemonade/OpenScPCA-analysis/pull/1036#discussion_r1952911292) since I think you actually meant neutrophils? I looked through the list of identified cell types here, not just the top ones shown in the plot, and those were all the myeloid cell types. 

Here's the updated report. 
[04-brain-and-cns-consensus-celltypes.nb.html.zip](https://github.com/user-attachments/files/18771509/04-brain-and-cns-consensus-celltypes.nb.html.zip)

jaclyn-taroni commented:
> I did add "neutrophils" and "myeloid leukocyte" to the myeloid group, but I didn't add granulocytes ([#1036 (comment)](https://github.com/AlexsLemonade/OpenScPCA-analysis/pull/1036#discussion_r1952911292)) since I think you actually meant neutrophils?

I meant granulocytes in the _general sense_ without any consideration for whether or not a cell could get annotated as an eosinophil, for example. Put another way, I wasn't paying attention to the potential consensus cell labels. If only neutrophil is possible, that works for me.

jaclyn-taroni commented on analyses/cell-type-consensus/exploratory-notebooks/04-brain-and-cns-consensus-celltypes.Rmd:
```diff
@@ -0,0 +1,404 @@
+
+This isn't super interesting to me since most classified cells are immune cells and there doesn't seem to be any major differences across disease types outside of ATRT having a lower percentage of classified cells than other projects. 
+
+## Macrophage vs. T cell populations 
```
```suggestion
## Myeloid vs. T cell populations 
```

jaclyn-taroni commented on analyses/cell-type-consensus/exploratory-notebooks/04-brain-and-cns-consensus-celltypes.Rmd:
```diff
@@ -0,0 +1,404 @@
+
+## Macrophage vs. T cell populations 
+
+It looks like there may be some differences between T cell and macrophage infiltrate based on the tumor type. 
+Here we will look specifically at all T cells and anything in the macrophage/monocyte lineage. 
+I'm including any cell types that are in our list of possible cell types and a child of [mononuclear phagocyte](https://www.ebi.ac.uk/ols4/ontologies/cl/classes/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FCL_0000113?lang=en) in the macro/mono group. 
+We should also note that "macrophage" here probably refers to microglia instead. 
```
This text could use a rewrite now.

jaclyn-taroni commented on analyses/cell-type-consensus/exploratory-notebooks/04-brain-and-cns-consensus-celltypes.Rmd:
````diff
@@ -0,0 +1,404 @@
+stacked_barchart(hgg_lgg_df, fill_column = "immune_broad_category", facet_variable = "hgg_vs_lgg", celltype_colors = immune_category_colors)
+```
+
+Consistent with the previous findings it looks like HGG tend to have higher T cell infiltrate while LGG tend to have higher macrophage infiltrate. 
````
```suggestion
Consistent with the previous findings it looks like HGG tend to have higher T cell infiltrate while LGG tend to have higher myeloid cell proportions. 
```

Would it be fair to characterize microglia as "infiltrate?"

jaclyn-taroni commented on analyses/cell-type-consensus/exploratory-notebooks/04-brain-and-cns-consensus-celltypes.Rmd:
````diff
@@ -0,0 +1,404 @@
+  theme(strip.text = element_text(size = 6)) 
+```
+
+- As expected based on the previous plot, glioblastoma appears to have greater T cell infiltrate than macrophages across all samples. 
+- There appears to be a handful of samples in the "High-grade glioma" group that also have high T cell infiltrate. 
+Notably, glioblastoma is a type of HGG. 
+- Generally, the T cell population seems lowest in LGG. 
+Ganglioglioma also has fewer samples with high percentage of T cell infiltrate and this is also a LGG. 
+- It looks like about 50% of the pilocytic astrocytomas have more T cell infiltrate than macrophages. 
````
Same comment about lightly rewriting this section.

jaclyn-taroni reviewed (approved):
I'm going to approve, since I don't think I need to see this again, but the text in the section about myeloid vs. T lineage should get rewritten before this is merged.

allyhawkins commented:
Just noting that I updated the text in https://github.com/AlexsLemonade/OpenScPCA-analysis/pull/1036/commits/b73b573ede96f6712920e4af66eaa8786c98e154.
